milli/src/documents/primary_key.rs

```diff
@@ -166,7 +166,7 @@ pub fn validate_document_id_value(document_id: Value) -> StdResult<String, UserE
             Some(s) => Ok(s.to_string()),
             None => Err(UserError::InvalidDocumentId { document_id: Value::String(string) }),
         },
-        Value::Number(number) if number.is_i64() => Ok(number.to_string()),
+        Value::Number(number) if !number.is_f64() => Ok(number.to_string()),
```

Comment:
Why did you change the condition from `is_i64` to `is_f64`?

Reply:
The `Number` type in JSON can be a float, so without the above guard clause, the document id would allow the character `.`.

Reply:
Oh yes I initially missed the new `!`, that’s a neat addition thanks